rllib/utils/checkpoints.py

```diff
@@ -497,32 +633,48 @@ def get_checkpoint_info(checkpoint: Union[str, Checkpoint]) -> Dict[str, Any]:
         info.update(
             {
                 "format": format,
-                "state_file": state_file,
+                "state_file": str(state_file),
             }
         )
 
         # Collect all policy IDs in the sub-dir "policies/".
-        policies_dir = os.path.join(checkpoint, "policies")
-        if os.path.isdir(policies_dir):
+        policies_dir = checkpoint / "policies"
+        if policies_dir.is_dir():
             policy_ids = set()
-            for policy_id in os.listdir(policies_dir):
-                policy_ids.add(policy_id)
+            for policy_id in policies_dir.iterdir():
+                policy_ids.add(policy_id.name)
             info.update({"policy_ids": policy_ids})
 
+        # Collect all module IDs in the sub-dir "learner/module_state/".
+        modules_dir = (
+            checkpoint
+            / COMPONENT_LEARNER_GROUP
+            / COMPONENT_LEARNER
+            / COMPONENT_RL_MODULE
+        )
+        if modules_dir.is_dir():
+            module_ids = set()
+            for module_id in modules_dir.iterdir():
+                # Only add subdirs (those are the ones where the RLModule data
+                # is stored, not files (could be json metadata files).
+                if (modules_dir / module_id).is_dir():
```

Comment:
Now a non-existent case but it near future possibly used: if an `RLModule` holds other `RLModule`s which hold other`RLModule`s (e.g. for different situations different modules for each agent in a MA setup) this will not work anymore.